src/TLuaInterpreter.cpp

```diff
@@ -12029,6 +12029,41 @@ bool TLuaInterpreter::compile(const QString & code, QString & errorMsg )
     }
 
     int error = luaL_dostring( L, code.toLatin1().data() );
+
+    QString n;
+    if( error != 0 )
+    {
+        string e = "Lua syntax error:";
+        if( lua_isstring( L, 1 ) )
+        {
+            e.append( lua_tostring( L, 1 ) );
+        }
+        errorMsg = "<b><font color='blue'>";
+        errorMsg.append( e.c_str() );
+        errorMsg.append("</b></font>");
+        if( mudlet::debugMode ){ TDebug(QColor(Qt::white),QColor(Qt::red))<<"\n "<<e.c_str()<<"\n">>0;}
+    }
+    else
+    {
+        if( mudlet::debugMode ){ TDebug(QColor(Qt::white),QColor(Qt::darkGreen))<<"\nLUA: code compiled without errors. OK\n" >> 0;}
+    }
+    lua_pop( L, lua_gettop( L ) );
+
+    if( error == 0 ) return true;
+    else return false;
+}
+
+bool TLuaInterpreter::compile(const QString & code, QString & errorMsg, const QString & name)
```

Comment:
It's not happening in compile, it's happening when I use prepend, which does "Prepends the string str to the beginning of this string and returns a reference to this string." per the Qt documentation. It's just that in this case, it's being passed the reference of the name itself, instead of an rvalue QString which isn't persisting beyond that line. I'll redo that bit of code so that it is safe to use with references as well, but I'll also do a PR because I see 0 reason why TTimer should be an exception and returns a reference instead of an rvalue.